### ISSUE TITLE
Support for accentuation?

### ISSUE THREAD

**Issue by zomars** [kortina/vscode-markdown-notes#47]
Coming downstream from the [Foam](https://github.com/foambubble/foam/issues/48) repo.

**Describe the bug**

When using accents like this:

```
- [[Seríes matemáticas]]
```

It replaces the accentuated letter with a dash.

**To Reproduce**

1. On any note create a new wiki link with an accent Ex. [[Seríes matemáticas]]

**Expected behavior**
It is expected to the new file to be named: `series-matematicas.md`.

**Screenshots**

![image](https://user-images.githubusercontent.com/3504472/86479297-1a311100-bd09-11ea-9b03-47ea9f36ba0b.png)

**Versions**
 - OS: macOS 10.15.5
 - VS Code Version: 1.46.1
 - Extension Version: 0.0.10

**Comment by kortina** [comment]
This results from: https://github.com/kortina/vscode-markdown-notes/blob/4af718ecf037791507d0d67d59f97ee5b961cccb/src/NoteWorkspace.ts#L163

We are replacing all non-word characters with the `-`. Is there an accent-character-friendly regex we can use instead of `\W` that will do the trick? Or do we need to enumerate every accent character? Anyone know?

My first inclination was to do something like this: https://stackoverflow.com/questions/990904/remove-accents-diacritics-in-a-string-in-javascript/37511463#37511463 but it destroys information and I think we might want to leave the accent characters in the filenames.

**Comment by zomars** [comment]
I wouldn't mind losing the accents. I think it's cleaner and URL friendly. But I'm speaking for myself here.

**Comment by jevakallio** [comment]
Hey @kortina! 

> It destroys information and I think we might want to leave the accent characters in the filenames.

I agree, we should leave non-word non-latin characters in the file names to support other languages. Here's another user with related issue in Japanese character:

https://github.com/foambubble/foam/issues/73

If people want to create url-friendly names for their own note titles, they can do that in their MD->HTML pipeline. "Convenience" here is very inconvenient for everyone from non-English speaking world, including my name :)

**Comment by digiguru** [comment]
Have a fix incoming

**Comment by digiguru** [comment]
I've added a proposed PR #53 above. I've unaccented, and then removed standard ASCII punctuation, but kept any non-latin characters as is. I think this is the most predictable default without introducing a setting. Happy to be challenged.

I won't start a fix for [foam bubble](https://github.com/foambubble/foam/issues/73) unless this one gets merged, as behaviour in there is affected by behaviour of `vscode-markdown-notes`

I _feels_ as if we could to pull the renaming logic into a dedicated node module. I've not got existing experience around that, but could always have a look if there's appetite.
